pybop/plotting/plot2d.py

```diff
@@ -64,11 +64,11 @@ def plot2d(
         cost = cost_or_optim
         plot_optim = False
 
-    if len(cost.parameters) < 2:
+    if hasattr(cost, "parameters") and len(cost.parameters) < 2:
```

Comment:
again, I think it would be clearer to use `isinstance(cost, pybamm.BaseCost`. The problem is, what if the user passes their own class that looks like a callable function, and it happens to have a `parameters` attribute?